src/Native/System.Security.Cryptography.Native/pal_cms.h

```diff
@@ -61,3 +61,49 @@ extern "C" int CryptoNative_CmsGetRecipientStackFieldCount(CmsRecipientStack* re
 Shims the CMS_RecipientInfo_free method
 */
 extern "C" void CryptoNative_CmsRecipientDestroy(CMS_RecipientInfo* recipient);
+
+/*
+Initializes a CMS_ContentInfo with the cipher given by the oid in algorithmOid and returns a pointer
+to it. 
+
+Status is set to 1 on success, 0 on unsupported algorithm, and -1 on null input.
+*/
+extern "C" CMS_ContentInfo* CryptoNative_CmsEncryptInitialize(ASN1_OBJECT* algorithmOid, int32_t* status);
+
+/*
+Adds all the originator certificates to the CMS_ContentInfo structure.
+
+Returns 1 on success, 0 on OpenSSL failure, and -1 on null input.
+*/
+extern "C" int CryptoNative_CmsAddOriginatorCerts(CMS_ContentInfo* cms, X509Stack* originatorCerts);
+
+/*
+Adds all the RecipientInfos of type IssuerAndSerialNumber to the CMS_ContentInfo structure.
```

Comment:
IssuerAndSerialNumber isn't a type of RecipientInfo.

Something like

Creates a KeyTransportRecipientInfo using the IssuerAndSerial identification method for each of the provided certificates.

Perhaps?